### PR TITLE
add key system flow design sections and update some design diagrams

### PR DESCRIPTION
**Pull request by cathyhongzhang**
Signed-off-by: Cathy Hong Zhang <hzhang@futurewei.com>

**What type of PR is this?**
> /kind design
> /kind documentation

**What this PR does / why we need it**:
Add key system flow design sections and update some design diagrams

### CODE REVIEW

File: docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md [hunk 1]
```diff
@@ -374,43 +373,114 @@ In our architecture, each scheduler has global view of resources and runs its sc
...
+All the security authentication/authorization as well as network error and retry during the communication with an external cluster, which may have varied latency, will be handled by the Dispatcher. If after a pre-defined number of retries, the Dispatcher still fails to send the VM/Container request to the selected cluster, the Dispatcher will put this VM/Container request back to the head of the Scheduling Work Queue and remove this cluster from the candidate list. 
+The Dispatcher will send the VM/Container request to the cluster via the VM/Container creation/deletion API over gRPC or HTTP2.   
+
+## 3.3 Scheduling Status Processing Flow
+This refers to the flow from Cluster->VM/Container Status Watcher->API Server->VM/Container Info DB. There are two ways to design and implement the scheduling status processing flow as described in the following two sections.
```
Comment (allison-wangnannan) [inline docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md:444]: I think  we don't  need to flow the status of VM/container.
The status change of  instances  can reflect by the resources status.

File: docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md [hunk 2]
```diff
@@ -374,43 +373,114 @@ In our architecture, each scheduler has global view of resources and runs its sc
...
+
+## 3.4 Cluster Resource Collection Flow
+This refers to the flow from Cluster->Cluster Resource Collector->API Server->Dynamic Resource Infor Cache DB. Similar to what is described in section 3.3, there are two ways to design and implement the resource collection flow as described in the following two sections. 
+
+### 3.4.1 Push Model (Preferred Model)
+This model requires the cluster to proactively send its resource information to the global scheduling framework’s Cluster Resource Collector. The cluster will only send its resource information when there is a resource change. If network traffic is a concern, “batch operation” can be used, i.e. resource update will be accumulated for a pre-defined interval and then sent to the global scheduling framework. This prevents injecting a burst of “update” traffic into the control plane’s network when the resource change frequency is very high at some time period. This interval value should be defined to keep a good balance between “accurate in-sync of global scheduling framework’s Dynamic Resource Info Cache with each cluster’s actual resource info DB” and “low network traffic and database/cache access traffic”
```
Comment (allison-wangnannan) [inline docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md:463]: The traffic flow need to be considered  independently,it's another topic which is very different from resource scheduler.

**Comment by CoderKevinZhang** [comment]
Looks good in general to me

**Review by CoderKevinZhang — APPROVED** [review]
Looks good in general to me

**Review by CoderKevinZhang — APPROVED** [review]
looks good to me

File: docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md (hunk 1 above)
Comment (cathyhongzhang) [reply docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md:444]: As discussed in yesterday's meeting, the global scheduler does not handle scheduling success/failure. So this status processing flow will be removed.

File: docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md (hunk 2 above)
Comment (cathyhongzhang) [reply docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md:463]: As discussed in the open source meeting, we will go for the push mode flow. In this mode, the cluster resource agent has knowledge about when and how often the resource changes in each cluster and can better decide when to send an update of resource to the global scheduling framework. We will use this push mode flow.

**Review by XiaoningDing — APPROVED** [review]

**Pull request merged**